### ISSUE TITLE
Unable to resolve service for type 'Microsoft.EntityFrameworkCore.Diagnostics.IDiagnosticsLogger`1[Microsoft.EntityFrameworkCore.DbLoggerCategory+Scaffolding ]' while attempting to activate 'ErikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory'.

### ISSUE THREAD

**Issue by Rotbarsch** [ErikEJ/EFCorePowerTools#1261]
Hello there,

trying to scaffold a DbContext from a .sqlproj or .dacpac file with the newest packages, including the handlebars package by Tony Sneed, throws the following error.

### Steps to reproduce

This is the command I call:

```
dotnet ef dbcontext scaffold `
"$dacpacProjectFile" ErikEJ.EntityFrameworkCore.SqlServer.Dacpac `
--context MATDBContext `
-o Entities `
--schema mat `
--context-dir . `
--data-annotations `
--force
```

This produces the following output:

```
Build started...
Build succeeded.
System.InvalidOperationException: Unable to resolve service for type 'Microsoft.EntityFrameworkCore.Diagnostics.IDiagnosticsLogger`1[Microsoft.EntityFrameworkCore.DbLoggerCategory+Scaffolding
]' while attempting to activate 'ErikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean 
throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteC
hain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean 
throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteC
hain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.EntityFrameworkCore.Design.Internal.DatabaseOperations.ScaffoldContext(String provider, String connectionString, String outputDir, String outputContextDir, String dbContextCla
ssName, IEnumerable`1 schemas, IEnumerable`1 tables, String modelNamespace, String contextNamespace, Boolean useDataAnnotations, Boolean overwriteFiles, Boolean useDatabaseNames, Boolean supp
ressOnConfiguring, Boolean noPluralize)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.ScaffoldContextImpl(String provider, String connectionString, String outputDir, String outputDbContextDir, String dbContextClassNa
me, IEnumerable`1 schemaFilters, IEnumerable`1 tableFilters, String modelNamespace, String contextNamespace, Boolean useDataAnnotations, Boolean overwriteFiles, Boolean useDatabaseNames, Bool
ean suppressOnConfiguring, Boolean noPluarlize)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.ScaffoldContext.<>c__DisplayClass0_0.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Unable to resolve service for type 'Microsoft.EntityFrameworkCore.Diagnostics.IDiagnosticsLogger`1[Microsoft.EntityFrameworkCore.DbLoggerCategory+Scaffolding]' while attempting to activate 'E
rikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory'.
```

### Further technical details

EF Core version in use: EF Core 6.0.1

EF Core Power Tools version: ErikEJ.EntityFrameworkCore.SqlServer.Dacpac 6.0.1-preview1 (also tried the latest stable, same result)

Database engine: MS SQL Server via dacpac

EntityFrameworkCore.Scaffolding.Handlebars: 6.0.3

**Comment by ErikEJ** [comment]
>> also tried the latest stable, same result

What do you mean by that specifically?

**Comment by ErikEJ** [comment]
I think the DesignTimeServices must be updated for EF Core 6: https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Design/Internal/SqlServerDesignTimeServices.cs

**Comment by Rotbarsch** [comment]
> > > also tried the latest stable, same result
> 
> What do you mean by that specifically?

I tried version 5.0.4 and 6.0.1-preview1. Both yielded the same results.


> I think the DesignTimeServices must be updated for EF Core 6: https://github.com/dotnet/efcore/blob/main/src/EFCore.SqlServer/Design/Internal/SqlServerDesignTimeServices.cs

Is there anything I can do about this.

**Comment by ErikEJ** [comment]
Wonder why it fails for 5.0 - can you share your .csproj?

**Comment by ErikEJ** [comment]
Reminder: Inject logger:

 private readonly IDiagnosticsLogger<DbLoggerCategory.Scaffolding> _logger;
        public SqlServerDacpacStoredProcedureModelFactory(IDiagnosticsLogger<DbLoggerCategory.Scaffolding> logger)
        {
            _logger = logger;
        }

**Comment by Rotbarsch** [comment]
Here you go:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="EntityFrameworkCore.Scaffolding.Handlebars" Version="6.0.3" />
    <PackageReference Include="ErikEJ.EntityFrameworkCore.SqlServer.Dacpac" Version="5.0.4" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="6.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Proxies" Version="6.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="6.0.1" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="6.0.1">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>

</Project>
```

**Comment by ErikEJ** [comment]
Ah, mixing packages like that will not work, 5.0.4 is for EF Core 5 only.

I think I just need to update the design time services, or accept a PR to that effect.

**Comment by Rotbarsch** [comment]
So the 6.0.1-preview1 package should work, in your opinion?

**Comment by ErikEJ** [comment]
>> So the 6.0.1-preview1 package should work, in your opinion?

No, it needs an update of the DesignTimeServices.

**Comment by Rotbarsch** [comment]
Oh, I see. A quick test of referencing a local copy of your repository and adding the IDiagnosticsLogger to the constructor seemed to solve this specific problem; but sadly a new error popped up complaining about being ```
Unable to resolve service for type 'ErikEJ.EntityFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactoryOptions' while attempting to activate 'ErikEJ.Enti
tyFrameworkCore.SqlServer.Scaffolding.SqlServerDacpacDatabaseModelFactory'.
```

So, I think it's best to wait for your fix. I'm looking forward to it and thanks in advance, you're doing a great job.

**Comment by ErikEJ** [comment]
I have published the 6.0.1-preview2 package that should fix this.
